### PR TITLE
#N8c834Zu Sign In users

### PR DESCRIPTION
**Pull request by Ntwali-Josue**
**What does this PR do**
This PR contains a backend work that allows users of phantom to login to the platform.

**Description of tasks to be completed**
The user should be able to login to the app by providing their email & password.
The API should return a JW token on a successful login.

**how should this be manually tested**
run `https://eagles-phantom-backend.herokuapp.com/v1/auth/login` in postman

**Other related stories**
https://trello.com/c/N8c834Zu/8-sign-in-to-phantom

**screenshot**
image

**Addition**
Access swagger documentation by running :
`https://eagles-phantom-backend.herokuapp.com/api-docs/`

### CODE REVIEW

**Review by Mugema123 — APPROVED** [review]

**Review by TuyizeeAnastase — APPROVED** [review]

**Review by Ally4 — COMMENTED** [review]
it`s not woks, no matter what I`m putting is it tells me the wrong email

**Comment by Ntwali-Josue** [comment]
@Ally4 you need to use the saved email which is Josh@gmail.com and the password admin

**Comment by Ally4** [comment]
I would like to ask you to edit the title of the PR too

**Review by Ally4 — APPROVED** [review]

**Review by Mugema123 — APPROVED** [review]

File: src/database/models/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const fs = require('fs');
+const path = require('path');
+const Sequelize = require('sequelize');
+const envConfigs = require('../config/config');
+
+const basename = path.basename(__filename);
+const env = process.env.NODE_ENV || 'development';
```
Comment (ceelogre) [inline src/database/models/index.js:7]: Is there any particular reason this is using `require` ?

File: src/languages/locales/fr.json [hunk 2]
```diff
@@ -1,4 +1,7 @@
 {
     "200":"200",
-    "Welcome to phantom!.":"bienvenue chez phantom!."
+    "Welcome to phantom!.":"bienvenue chez phantom!.",
+    "logged In successfull": "Connexion réussie",
+	"wrong email or password": "mauvais email ou mot de passe",
+	"Invalid email or password": "email ou mot de passe invalide"
```
Comment (ceelogre) [inline src/languages/locales/fr.json]: These can be merged into one message.

File: src/test/test.js [hunk 3]
```diff
@@ -1,19 +1,31 @@
...
+  it('check user info', (done) => {
+    chai
+      .request(app)
+      .post('/v1/auth/login')
+      .end((err, res) => {
+        expect(res).to.have.status(401);
+        expect(res.body.message).to.equals('wrong email or password');
         done();
       });
   });
```
Comment (ceelogre) [inline src/test/test.js]: Add (at least) one more test for the case where a user provides the right email and password.

**Review by ceelogre — CHANGES_REQUESTED** [review]
1. Check out the comments I've added
2. Add the right `Accept Language`
3. Show log messages only in `dev` mode.

File: src/database/models/index.js (hunk 1 above)
Comment (Ntwali-Josue) [reply src/database/models/index.js:7]: Because the models are created automatically, they come in es5

**Review by ceelogre — CHANGES_REQUESTED** [review]
This pr doesn't have `.env.example`

**Review by ceelogre — APPROVED** [review]

**Pull request merged**